### PR TITLE
Misuv 8059 remove fullfilled obligations call and rename obligations model classes

### PR DESCRIPTION
**Pull request by joesheridan**

### CODE REVIEW

File: app/models/obligations/SingleObligationModel.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * limitations under the License.
+ */
+
+package models.obligations
+
+import play.api.libs.json.{Format, Json}
+
+import java.time.LocalDate
+
+case class SingleObligationModel(start: LocalDate,
```
[the remark this answers is not in the document]
Comment (joesheridan) [reply app/models/obligations/SingleObligationModel.scala:23]: In the past, (some) developers have not been very good at differentiating between Things and Thing i.e. regularly getting these two confused - that's why I wanted to make it super clear. I agree about the suffix but the 'model' suffix was already there in BE and FE and I didn't want to rock the boat too much.

File: app/models/obligations/GroupedObligationsModel.scala [hunk 2]
```diff
@@ -14,12 +14,15 @@
...
  */
 
-package models.nextUpdates
+package models.obligations
 
-import java.time.LocalDate
+import play.api.libs.json.{Format, Json}
 
-sealed trait NextUpdateStatus
+case class GroupedObligationsModel(identification: String, obligations: List[SingleObligationModel]) {
```
[the remark this answers is not in the document]
Comment (joesheridan) [reply app/models/obligations/GroupedObligationsModel.scala]: What would be really nice is just to have a flat list of obligations 🤔

[the remark this answers is not in the document]
Comment (ian-hmrc) [reply app/models/obligations/GroupedObligationsModel.scala:21]: OK. But would the rename make sense in the mean-time?

File: app/models/obligations/SingleObligationModel.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (joesheridan) [reply app/models/obligations/SingleObligationModel.scala:23]: As mentioned before, the grouping is not always by income source, so this name wouldn't be appropriate imho.

File: app/models/obligations/GroupedObligationsModel.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (joesheridan) [reply app/models/obligations/GroupedObligationsModel.scala:21]: This intermediary class will be removed in any case as part of the future changes discussed. I think the name is appropriate for now.

[the remark this answers is not in the document]
Comment (ian-hmrc) [reply app/models/obligations/GroupedObligationsModel.scala:21]: OK.

File: app/models/obligations/SingleObligationModel.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (ian-hmrc) [reply app/models/obligations/SingleObligationModel.scala:23]: OK, in that case it's all good. :)

**Pull request merged**